packages/esm-patient-forms-app/src/forms/forms-see-all.tsx

```diff
@@ -0,0 +1,128 @@
+import React from 'react';
+import dayjs from 'dayjs';
+import first from 'lodash-es/first';
+import ContentSwitcher from 'carbon-components-react/es/components/ContentSwitcher';
+import Switch from 'carbon-components-react/es/components/Switch';
+import FormView from './form-view-see-all.component';
+import styles from './forms.component.scss';
+import { EmptyState, ErrorState } from '@openmrs/esm-patient-common-lib';
+import { useTranslation } from 'react-i18next';
+import { fetchAllForms, fetchPatientEncounters } from './forms.resource';
+import { filterAvailableAndCompletedForms } from './forms-utils';
+import { Encounter, Form } from '../types';
+import EmptyFormView from './empty-form.component';
+import { getObjectFHIR } from './get-FHIR-object';
+
+enum FormViewState {
+  recommended = 0,
+  completed,
+  all,
+}
+
+const Forms = () => {
+  const urlPathArray = window.location.pathname.split('/');
+  const pathUuid = urlPathArray[urlPathArray.length - 2];
+  const [patientUuid, setUUID] = React.useState(null);
+  const [patient, setPatient] = React.useState<fhir.Patient>();
+
+  React.useEffect(() => {
+    getObjectFHIR(patientUuid).then(setPatient);
+    setUUID(pathUuid);
+  }, [pathUuid, patientUuid]);
+
+  const { t } = useTranslation();
+  const displayText = t('forms', 'Forms');
+  const headerTitle = t('forms', 'Forms');
+  const [error, setError] = React.useState(null);
+  const [forms, setForms] = React.useState<Array<Form>>([]);
+  const [encounters, setEncounters] = React.useState<Array<Encounter>>([]);
+  const [completedForms, setCompletedForms] = React.useState<Array<Form>>([]);
+  const [selectedFormView, setSelectedFormView] = React.useState<FormViewState>(FormViewState.all);
+  const [filledForms, setFilledForms] = React.useState<Array<Form>>([]);
+
+  React.useEffect(() => {
+    fetchAllForms().subscribe((forms) => setForms(forms), setError);
```

Comment:
What's that with `subscribe`? Why don't we `unsubscribe`? Why no simple `openmrsFetch`?

Please avoid using these observables.

Reply:
This file is a adaptation of [forms.component.tsx](https://github.com/openmrs/openmrs-esm-patient-chart/blob/master/packages/esm-patient-forms-app/src/forms/forms.component.tsx) and I have only made minor alterations to it to and have not tried to create the widget from scratch. Most of the code directly comes from the original file which was used to build the Form Entry widget and is now a part of the "openmrs-esm-patient-chart" repo.

Reply:
We should reuse components - not copy them. This is a no go that results in barely maintainable code that is bloated and contains bugs.

Reply:
Exactly my point here! It was impossible to **reuse** components here. Let me explain the origins of this bug. If you go to [form-view.component.tsx](https://github.com/openmrs/openmrs-esm-patient-chart/blob/master/packages/esm-patient-forms-app/src/forms/form-view.component.tsx) and observe Ln #169, the variable "pageUrl" has been hardcoded. Then again on Ln #170 the variable "pageSize" has been **hardcoded**. This was the key component required in this situation and it was impossible re-use it. So I had to go for the closest work around. The other option is rewrting everything which is clearly beyond the scope of this ticket. Now, we need to understand why reusable components with hard coded variables are part of our repo in the first place and make corrections there.

Reply:
But putting the old codes aside, I will rewrite certain aspects. I will make changes based on your suggestions and come up with a new commit. Thank you.

Reply:
This is done and I have replaced the older function with "openmrsFetch"